riscv/tests/riscv.rs

```diff
@@ -1,21 +1,53 @@
 mod common;
 
-use std::path::PathBuf;
-
 use common::verify_riscv_asm_string;
 use mktemp::Temp;
+use powdr_backend::BackendType;
 use powdr_number::GoldilocksField;
 use powdr_pipeline::{
     test_util::{verify_asm_string, verify_pipeline},
-    Pipeline,
+    verify::verify,
+    Pipeline, Stage,
 };
+use std::path::PathBuf;
 use test_log::test;
 
 use powdr_riscv::{
     continuations::{rust_continuations, rust_continuations_dry_run},
     CoProcessors,
 };
 
+/// Compiles and runs a rust file with continuations, runs the full
+/// witness generation & verifies it using Pilcom.
+pub fn test_continuations(case: &str) {
+    let rust_file = format!("{case}.rs");
+    let coprocessors = CoProcessors::base().with_poseidon();
+    let temp_dir = Temp::new_dir().unwrap();
+    let riscv_asm =
+        powdr_riscv::compile_rust_to_riscv_asm(&format!("tests/riscv_data/{rust_file}"), &temp_dir);
+    let powdr_asm = powdr_riscv::compiler::compile(riscv_asm, &coprocessors, true);
+
+    // Manually create tmp dir, so that it is the same in all chunks.
+    let tmp_dir = mktemp::Temp::new_dir().unwrap();
+
+    let pipeline_factory = || {
+        Pipeline::<GoldilocksField>::default()
+            .from_asm_string(powdr_asm.clone(), Some(PathBuf::from(&rust_file)))
+            .with_prover_inputs(Default::default())
+            .with_output(tmp_dir.to_path_buf(), false)
+    };
+    let pipeline_callback = |pipeline: Pipeline<GoldilocksField>| -> Result<(), ()> {
+        // Can't use `verify_pipeline`, because the pipeline was renamed in the middle of after
+        // computing the constants file.
```

Comment:
Not sure I get this comment

Reply:
`verify_pipeline` looks at the pipeline name and uses that to find all the artifacts. But the continuations code runs until all fixed columns are persisted and then branches off to run all chunks independently. The pipeline is renamed then (so that artifacts don't collide).

Before this PR, we used to copy `many_chunks_constants.bin` to `many_chunks_chunk_0_constants.bin` etc, to work around this. I think now it's cleaner.

Reply:
ah right